app/src/main/java/com/example/concordia_campus_guide/Activities/RoutesActivity.java

```diff
@@ -62,9 +58,12 @@ protected void onCreate(Bundle savedInstanceState) {
         // set back button
         setBackButtonOnClick();
 
-        // get all possible routes. The following statements should happen on an onClick event.
-        String url = UrlBuilder.build(new LatLng(mViewModel.getFrom().getCenterCoordinates().getLatitude(), mViewModel.getFrom().getCenterCoordinates().getLongitude()), new LatLng(mViewModel.getTo().getCenterCoordinates().getLatitude(), mViewModel.getTo().getCenterCoordinates().getLongitude()), ClassConstants.TRANSIT);
-        new DirectionsApiDataRetrieval(RoutesActivity.this).execute(url);
+        //null check
+        if(mViewModel.getFrom().getCenterCoordinates() != null && mViewModel.getTo().getCenterCoordinates() != null){
+            // get all possible routes. The following statements should happen on an onClick event.
+            String url = UrlBuilder.build(new LatLng(mViewModel.getFrom().getCenterCoordinates().getLatitude(), mViewModel.getFrom().getCenterCoordinates().getLongitude()), new LatLng(mViewModel.getTo().getCenterCoordinates().getLatitude(), mViewModel.getTo().getCenterCoordinates().getLongitude()), ClassConstants.TRANSIT);
+            new DirectionsApiDataRetrieval(RoutesActivity.this).execute(url);
```

Comment:
do you mind refactoring it and making them into separate variables then pass them to the function? Because it is really hard to read it like this.

Reply:
I'm just here to do the bug fix, @ferdousara-parvin can you fix as part of your story since this is your story?

Reply:
Doesn't this go against the principle of having all the data in the viewmodel?

Reply:
yeah sorry I am aware this is hard to read. I fixed it in my branch, will push it soon. And I don't get your last comment @thomasgauvin